### PR TITLE
improvement/multi stage docker and python 3.11

### PR DESCRIPTION
**Pull request by dabeeeenster**
Multi stage build for python venv too

This also gets python 3.11 working!

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,38 +11,45 @@ ENV STATIC_ASSET_CDN_URL=/static/
...
+COPY api /app/
 
 # Install SAML dependency if required
 ARG SAML_INSTALLED="0"
 RUN if [ "${SAML_INSTALLED}" = "1" ]; then apt-get update && apt-get install -y xmlsec1; fi;
 
-# Install python dependencies
-RUN pip install -r requirements.txt --no-cache-dir --compile
+# Install postgres
+RUN apt-get update && apt-get install -y libpq-dev && rm -rf /var/lib/apt/lists/*
```
Comment (matthewelwell) [inline Dockerfile]: Do we want to only do this if target arch is arm?

File: Dockerfile [hunk 2]
```diff
@@ -11,38 +11,46 @@ ENV STATIC_ASSET_CDN_URL=/static/
...
 
 # Install SAML dependency if required
 ARG SAML_INSTALLED="0"
 RUN if [ "${SAML_INSTALLED}" = "1" ]; then apt-get update && apt-get install -y xmlsec1; fi;
 
-# Install python dependencies
-RUN pip install -r requirements.txt --no-cache-dir --compile
+# arm architecture platform builds need postgres drivers installing via apt
+ARG TARGETARCH
+RUN if [ "${TARGETARCH}" != "amd64" ]; then apt-get update && apt-get install -y gcc libpq-dev && rm -rf /var/lib/apt/lists/*; fi;
```
Comment (matthewelwell) [inline Dockerfile]: Do we need gcc here too?

File: api/Dockerfile [hunk 3]
```diff
@@ -1,35 +1,43 @@
...
-RUN apt update && apt install -y xmlsec1
+# Install SAML dependency if required
+ARG SAML_INSTALLED="0"
+RUN if [ "${SAML_INSTALLED}" = "1" ]; then apt-get update && apt-get install -y xmlsec1; fi;
 
-# Install python dependencies
-RUN pip install -r requirements.txt --no-cache-dir --compile
+# arm architecture platform builds need postgres drivers installing via apt
+ARG TARGETARCH
+RUN if [ "${TARGETARCH}" != "amd64" ]; then apt-get update && apt-get install -y gcc libpq-dev && rm -rf /var/lib/apt/lists/*; fi;
```
Comment (matthewelwell) [inline api/Dockerfile]: and here

File: Dockerfile (hunk 2 above)
Comment (dabeeeenster) [reply Dockerfile:42]: yes

File: api/Dockerfile (hunk 3 above)
Comment (dabeeeenster) [reply api/Dockerfile:28]: and 

Y
E
S

File: Dockerfile (hunk 2 above)
Comment (dabeeeenster) [reply Dockerfile]: removed

File: api/Dockerfile (hunk 3 above)
Comment (dabeeeenster) [reply api/Dockerfile]: removed

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**